File: PB - More Exercises Python/7.0 Nested Loops - More Exercises/10. Profit.py
one_coins, two_coins, five_coins, all_sum = [int(input()) for _ in range(4)]

for one in range(one_coins + 1):
    for two in range(two_coins + 1):
        for five in range(five_coins + 1):
            if one * 1 + two * 2 + five * 5 == all_sum:
                print(f'{one} * 1 lv. + {two} * 2 lv. + {five} * 5 lv. = {all_sum} lv.')




'''
3
2
3
10
'''

File: PB - More Exercises GO/7.0 Nested Loops - More Exercises/10. Profit.go
package main

import "fmt"

func main() {
	var oneCoins, twoCoins, fiveCoins, allSum int
	fmt.Scan(&oneCoins, &twoCoins, &fiveCoins, &allSum)

	for one := 0; one <= oneCoins; one++ {
		for two := 0; two <= twoCoins; two++ {
			for five := 0; five <= fiveCoins; five++ {
				if one*1+two*2+five*5 == allSum {
					fmt.Printf("%v * 1 lv. + %v * 2 lv. + %v * 5 lv. = %v lv.\n", one, two, five, allSum)
				}
			}
		}
	}

}

/*
name   :10. Profit
input  :3 2 3 10
output :
0 * 1 lv. + 0 * 2 lv. + 2 * 5 lv. = 10 lv.
1 * 1 lv. + 2 * 2 lv. + 1 * 5 lv. = 10 lv.
3 * 1 lv. + 1 * 2 lv. + 1 * 5 lv. = 10 lv.
*/
